backend/api/views/datamanager.py

```diff
@@ -180,7 +180,7 @@ def get_queryset(self):
         return DataManager.objects.filter(channel=get_channel_name(self.request))
 
     def get_serializer_class(self):
-        if self.action == "retrieve":
+        if self.action in ["list", "retrieve"]:
```

Comment:
I think you can use the same serializer as the extra fields are read only. Let's do some cleaning:
- add the `test_data_sample_keys` and `train_data_sample_keys` to the `DataManagerSerializer`
- remove `DataManagerWithRelationsSerializer` everywhere

Reply:
Note that `Datasample.test_only` is about to be deprecated and all of this are going to be removed.

Reply:
After looking a bit more, `DataManagerSerializer` is also used in `ComputeTaskInputSerializer` and  `ComputeTaskInputAssetSerializer`. So more views are impacted, e.g. "list input assets" or "retrieve compute task" would now include data manager train/test data samples.

This is a bigger change than is intended with this PR, especially since it introduces a performance hit on more routes than just the "list data managers" route.

So I see 3 routes from here:

1. Use `DataManagerWithRelationsSerializer` everywhere (consistent but expensive)
2. Use `DataManagerWithRelationsSerializer` for list/retrieve, and keep using `DataManagerSerializer` for other routes (this PR as of writing this comment)
3. Use `DataManagerWithRelationsSerializer` for "retrieve" only. Use `DataManagerSerializer` for other routes (`main` branch)
   - if we go with this option, we should use different models for data managers models in the SDK depending on the route (which is what you meant in [your comment](https://github.com/Substra/substra-backend/pull/515#discussion_r988746001)?). This would eliminate the situation described in this PR's description (there are `{test,train}_data_sample_keys` fields but they are empty, which is misleading.) This change on the SDK side doesn't look trivial though

I'm not fond of Option 1 as it would introduce a performance hit for little benefit. 

Option 2 looks good.

Option 3 leaves unresolved the issue that this PR tries to address. It could be addressed in the SDK instead, but it doesn't look trivial (or a good idea?) because the SDK assumes that "list" and "get" API routes return the same models. Changing that would involve introducing a special case for data managers:

https://github.com/Substra/substra/blob/c52f2f94907200fe550717cfb1541a887569da39/substra/sdk/backends/remote/backend.py#L45-L48

https://github.com/Substra/substra/blob/c52f2f94907200fe550717cfb1541a887569da39/substra/sdk/backends/remote/backend.py#L92-L99

What do you think?

Reply:
As we intend to deprecated `test_only` fields, I think it's not a bit deal and I let you see what you think is the best option. All of this should be refactored soon. Thanks for checking this.